processor/metricstransformprocessor/metrics_transform_processor.go

```diff
@@ -27,10 +29,42 @@ import (
 )
 
 type internalTransform struct {
-	MetricName string
-	Action     ConfigAction
-	NewName    string
-	Operations []internalOperation
+	MetricIncludeFilter internalFilter
+	Action              ConfigAction
+	NewName             string
+	Operations          []internalOperation
+}
+
+type internalFilter interface {
+	getMatches(toMatch metricNameMapping) []*metricspb.Metric
+}
+
+type internalFilterStrict struct {
+	include string
+}
+
+func (f internalFilterStrict) getMatches(toMatch metricNameMapping) []*metricspb.Metric {
+	if metrics, ok := toMatch[f.include]; ok {
```

Comment:
You can replace this whole func with: `return toMatch[f.include]`.

Reply:
This comment has become redundant as I refactored this function to support substition - see comment below